Tests/UnitTests/Identity/CustomerInfoManagerTests.swift

```diff
@@ -448,15 +418,45 @@ class CustomerInfoManagerTests: BaseCustomerInfoManagerTests {
     func testCacheCustomerInfoSendsToDelegateIfChanged() {
         self.customerInfoManager.cache(customerInfo: self.mockCustomerInfo, appUserID: "myUser")
         expect(self.customerInfoManagerChangesCallCount).toEventually(equal(1))
-        expect(self.customerInfoManagerLastCustomerInfo) == self.mockCustomerInfo
+        expect(self.customerInfoManagerLastCustomerInfoChange) == (old: nil, new: self.mockCustomerInfo)
+    }
+
+    func testCacheCustomerInfoSendsMultipleUpdatesIfChange() throws {
+        let newCustomerInfo = try CustomerInfo(data: [
+            "request_date": "2023-12-21T02:40:36Z",
+            "subscriber": [
+                "original_app_user_id": "new user",
+                "first_seen": "2019-06-17T16:05:33Z",
+                "subscriptions": [:] as [String: Any],
+                "other_purchases": [:] as [String: Any],
+                "original_application_version": NSNull()
+            ]  as [String: Any]
+        ])
+
+        self.customerInfoManager.cache(customerInfo: self.mockCustomerInfo, appUserID: "myUser")
+        self.customerInfoManager.cache(customerInfo: newCustomerInfo, appUserID: "myUser")
+
+        expect(self.customerInfoManagerChangesCallCount).toEventually(equal(2))
+        expect(self.customerInfoManagerLastCustomerInfoChange) == (old: self.mockCustomerInfo, new: newCustomerInfo)
     }
 
     func testCacheCustomerInfoSendsToDelegateWhenComputedOnDevice() {
         let info = self.mockCustomerInfo.copy(with: .verifiedOnDevice)
 
         self.customerInfoManager.cache(customerInfo: info, appUserID: "myUser")
         expect(self.customerInfoManagerChangesCallCount).toEventually(equal(1))
-        expect(self.customerInfoManagerLastCustomerInfo) == info
+        expect(self.customerInfoManagerLastCustomerInfoChange) == (old: nil, new: info)
+    }
+
+    func testCacheCustomerInfoSendsToDelegateAfterCachingComputedOnDevice() {
```

Comment:
The first implementation broke integration tests in a way that this test now captures.